tests/base/src/test/java/org/wildfly/security/audit/SyslogAuditEndpointTest.java

```diff
@@ -0,0 +1,301 @@
+        socketFactory = SocketFactory.getDefault();
+        udpServer = SimpleSyslogServer.createUdp(UDP_PORT);
+        tcpServer = SimpleSyslogServer.createTcp(TCP_PORT, false);
+    }
+
+    /**
+     * Safely shutdowns the server
+     */
+    @AfterClass
+    public static void shutdownServer() {
+        udpServer.close();
+        tcpServer.close();
+    }
+
+    /**
+     * Tests the syslog audit endpoint with RFC 3164 format and critical priority
+     */
+    @Test
+    public void testGenericRfc3164SyslogEvent() throws Exception {
+        AuditEndpoint endpoint = setupEndpointUdp(SyslogHandler.SyslogType.RFC3164);
+        endpoint.accept(EventPriority.CRITICAL, RFC3164_LOG_MESSAGE);
+        endpoint.close();
+        discardMessage(udpServer);
+        verifyUdpMessage(RFC3164_LOG_MESSAGE);
+    }
+
+    /**
+     * Tests the syslog audit endpoint with RFC 5424 format and critical priority
+     */
+    @Test
+    public void testGenericRfc5424SyslogEvent() throws Exception {
+        AuditEndpoint endpoint = setupEndpointUdp(SyslogHandler.SyslogType.RFC5424);
+        endpoint.accept(EventPriority.CRITICAL, RFC5424_LOG_MESSAGE);
+        endpoint.close();
+        discardMessage(udpServer);
+        verifyUdpMessage(RFC5424_LOG_MESSAGE);
+    }
+
+    /**
+     * Tests that the endpoint rejects a bad integer
+     */
+    @Test
+    public void testBadIntegerReconnectAttempts() throws Exception {
+        try {
+            // Since the endpoint gives a message upon creation there is no need to call .accept()
+            AuditEndpoint endpoint = setupEndpointBadUdpReconnectAttempts(BAD_RECONNECT_NUMBER);
+        } catch (IllegalArgumentException e) {
+            assertEquals(e.getMessage(), "ELY11008: The reconnect attempts value of " + BAD_RECONNECT_NUMBER + " is invalid. Please use an integer value >= -1.");
+        }
+    }
+
+    /**
+     * Tests that the server continuously attempts to resend the message for {@link SyslogAuditEndpointTest#RECONNECT_TIMEOUT} milliseconds
+     */
+    @Test
+    public void testFailureInfiniteReconnectAttempts() throws Exception {
+        Thread t = new Thread(() -> {
+            try {
+                // Since the endpoint gives a message upon creation there is no need to call .accept()
+                AuditEndpoint endpoint = setupEndpointBadUdpReconnectAttempts(-1);
+            } catch (Exception e) {
+                System.out.println("An unexpected error has occurred: " + e.getMessage());
+            }
+        });
+        t.start();
+        Thread.sleep(RECONNECT_TIMEOUT);
+        t.interrupt();
+    }
+
+    /**
+     * Tests that the server never reattempts to send the message
+     */
+    @Test
+    public void testFailureZeroReconnectAttempts() throws Exception {
+        try {
+            // Since the endpoint gives a message upon creation there is no need to call .accept()
+            AuditEndpoint endpoint = setupEndpointBadUdpReconnectAttempts(0);
```

Comment:
It looks like there should be an ```Assert.fail``` call after this line so that the test fails if no exception is thrown.

Reply:
Fixed